tiantianXia/assignments/cyclicSort/cyclicSortEasy/cyclicSortEasy.js

```diff
@@ -4,8 +4,39 @@
 
 // Write a function to sort the objects in-place on their creation sequence number in O(n) and without any extra space. For simplicity, let’s assume we are passed an integer array containing only the sequence numbers, though each number is actually an object.
 
+/**
+ * inputs: 
+ * @param {*} nums: an array that containing unique unsorted positive intergers.
+ * 
+ * outputs
+ * @returns nums: a sorted array
+ * 
+ * step 1: loop all the elements
+ * step 2: set j be the number of the nums[i] - 1 so that the function is able to compare the number at current position and number
+ *          in the j position which is the position that current element should at
+ * step 3: swap two element if two element is no equal
+ * 
+ * variable i: for iteration
+ * 
+ * while loop to traverse nums
+ *  set j as nums[i] - 1 so that we can compare that number at that position and swap
+ *    if current nums is not equal to the nums at j
+ *      swap current num and num at j
+ *    else increment i
+ * end loop
+ * return nums
+ * edge case nums[] => []
+ */
```

Comment:
This portion of overview can be something important to keep you going into the right direction as you code, BUT, it is so close to the actual code that I may consider it DRY. I would say this can be deleted once the main function is complete.